File: impl/java/src/util/Point.java
package util;

public class Point implements Comparable<Point> {
    private int x;
    private int y;
    private double angle;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
        double degrees = Math.toDegrees(Math.atan2(y, x));
        angle = degrees >= 0 ? degrees : degrees + 360;
    }

    private double distanceFromOrigin() {
        return Math.sqrt((double) (x * x + y * y));
    }

    public int compareTo(Point p) {
        if (angle == p.angle) {
            return (int) (distanceFromOrigin() - p.distanceFromOrigin());
        } else {
            return (int) Math.ceil(angle - p.angle);
        }
    }

    public String toString() {
        return String.format("(%d, %d, %.2f)", x, y, angle);
    }

    public boolean equals(Point p) {
        if (x == p.x && y == p.y) return true;
        return false;
    }

    public static void main(String[] args) {
        System.out.println(new Point(1, 1));
    }
}

File: impl/python/point.py
import math

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
        radians = math.atan2(y,x)
        degrees = math.pi * radians
        angle = math.nan
        if degrees >= 0:
            angle = degrees
        else:
            angle = degrees + 360
        self.angle = angle

    def distance_from_origin(self):
        return math.sqrt(self.x * self.x + self.y * self.y)

    def __lt__(self, other):
        if self.angle == other.angle:
            return self.distance_from_origin() < other.distance_from_origin()
        else:
            return self.angle < other.angle

    def __gt__(self, other):
        if self.angle == other.angle:
            return self.distance_from_origin() > other.distance_from_origin()
        else:
            return self.angle > other.angle
    
    def __le__(self, other):
        if self.angle == other.angle:
            return self.distance_from_origin() <= other.distance_from_origin()
        else:
            return self.angle <= other.angle
    
    def __ge__(self, other):
        if self.angle == other.angle:
            return self.distance_from_origin() >= other.distance_from_origin()
        else:
            return self.angle >= other.angle
    
    def __eq__(self, other):
        return self.angle == other.angle and self.distance_from_origin() == other.distance_from_origin()
    
    def __ne__(self, other):
        return self.angle != other.angle or self.distance_from_origin() != other.distance_from_origin()
    